mlos_bench/db/db_schema.sql

```diff
@@ -0,0 +1,83 @@
+--
+-- Copyright (c) Microsoft Corporation.
+-- Licensed under the MIT License.
+--
+-- DB schema for storing MLOS benchmarking results.
+-- The syntax works for SQLite3, DuckDB, PostgreSQL, and MySQL / MariaDB.
+
+DROP TABLE IF EXISTS trial_telemetry;
+DROP TABLE IF EXISTS trial_results;
+DROP TABLE IF EXISTS trial_config;
+DROP TABLE IF EXISTS experiment_merge;
+DROP TABLE IF EXISTS trial_status;
+DROP TABLE IF EXISTS experiment_config;
+
+CREATE TABLE experiment_config (
+    exp_id VARCHAR(255) NOT NULL,
+    descr TEXT,
+    git_repo VARCHAR(255) NOT NULL,
+    git_commit VARCHAR(40) NOT NULL,
+
+    PRIMARY KEY (exp_id)
+);
+
+CREATE TABLE trial_status (
+    exp_id VARCHAR(255) NOT NULL,
+    trial_id INTEGER NOT NULL,
+    ts_start TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    ts_end TIMESTAMP,
+    -- Should match the text IDs of `mlos_bench.environment.Status` enum:
+    status VARCHAR(16) NOT NULL,
+
+    PRIMARY KEY (exp_id, trial_id),
+    FOREIGN KEY (exp_id) REFERENCES experiment_config(exp_id)
+);
+
+CREATE TABLE experiment_merge (
+    dest_exp_id VARCHAR(255) NOT NULL,
+    dest_trial_id INTEGER NOT NULL,
+    source_exp_id VARCHAR(255) NOT NULL,
+    source_trial_id INTEGER NOT NULL,
+
+    UNIQUE (dest_exp_id, dest_trial_id, source_exp_id, source_trial_id),
+
+    FOREIGN KEY (dest_exp_id, dest_trial_id)
+        REFERENCES trial_status(exp_id, trial_id),
+
+    FOREIGN KEY (source_exp_id, source_trial_id)
+        REFERENCES trial_status(exp_id, trial_id)
+);
+
+CREATE TABLE trial_config (
```

Comment:
We may need one of these for tables for the experiment level.
For instance:
```
experiment_config:
  param_id: varchar
  param_type: enum
  param_default: varchar
  param_range: varchar
```

Reply:
Without that we'd have to have the config json on hand for us to be able to tell whether two configs were mergeable.

Reply:
let's do it in the next PR. I've added this table, but then figured that it requires an FK to it from `trial_config`, and that, in turn, requires the table to be pre-populated before we run any experiments, which adds extra logic to the unit test fixtures, etc., etc. 

How about we keep the DB schema to the absolute minimum in this PR and add the tunables metadata later when I implement the experiment merge?